### ISSUE TITLE
`DTZ007` false-positive when using time format from const

### ISSUE THREAD

**Issue by aw1cks** [astral-sh/ruff#7488]
Hi there, first of all just wanted to say thanks for this awesome project.

I noticed a bug with the `DTZ007` rule - it seems like it's not reading my const value and falsely telling me that I'm not using `%z`.
### Configuration

```shell
$ ruff --version
ruff 0.0.290
$ cat pyproject.toml 
[tool.ruff]
select = [
  "DTZ",
]
$ 
```

### Code example

This works fine:

```python
import datetime

my_date = "Thu, 14 Sep 2023 07:45:07 +0000"
mydt = datetime.datetime.strptime(my_date, "%a, %d %b %Y %H:%M:%S %z")
```

```shell
$ ruff .
$
```

However, when using a const:

```python
import datetime

MY_DATE_FORMAT = "%a, %d %b %Y %H:%M:%S %z"

my_date = "Thu, 14 Sep 2023 07:45:07 +0000"
mydt = datetime.datetime.strptime(my_date, MY_DATE_FORMAT)
```

```shell
$ ruff .
test.py:6:8: DTZ007 The use of `datetime.datetime.strptime()` without %z must be followed by `.replace(tzinfo=)` or `.astimezone()`
Found 1 error.
$
```

**Comment by aw1cks** [comment]
Whoops - I missed https://github.com/astral-sh/ruff/issues/1306 - closing
